model/SearchEngine/Driver/Elasticsearch/QueryBuilder.php

```diff
@@ -207,24 +214,66 @@ private function buildConditionsByType(string $type, array $blocks): array
         return $this->getResourceConditions($blocks);
     }
 
+    private function containsLogicalModifier(string $block) {
+        $found = false;
+        foreach(self::LOGIC_MODIFIERS as $key => $logicModifier) {
+            if(strpos($block, $logicModifier) !== false) {
+                $found = true;
+                break;
+            }    
+        }    
+        return $found;
+    }
```

Comment:
chore suggestion: 

- Pay attention to formatting as we need them between operators.
- You do not need tmp variables, you can just return the result directly.
- There is no need to have the `key` in the foreach if you do not use it

```suggestion
    private function containsLogicalModifier(string $block) {
        foreach (self::LOGIC_MODIFIERS => $logicModifier) {
            if (strpos($block, $logicModifier) !== false) {
                return true;
            }
        }

        return false;
    }
```